sys/arduino/SPI.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_arduino_api
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the Arduino 'SPI' interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#ifdef MODULE_PERIPH_SPI
+
+extern "C" {
+#include "assert.h"
+}
+
+#include "SPI.h"
+
+SPISettings::SPISettings(uint32_t clock, uint8_t bitOrder, uint8_t dataMode)
```

Comment:
The problem is that the parameter `clock` has same name as member `clock`.
```suggestion
SPISettings::SPISettings(uint32_t clk, uint8_t bitOrder, uint8_t dataMode)
```